d2/src/main/java/com/linkedin/d2/xds/XdsClient.java

```diff
@@ -107,6 +84,27 @@ public String getVersion()
     {
       return _version;
     }
+
+    @Override
+    public boolean equals(Object object)
```

Comment:
Hi, @bohhyang
So, the reason why the parameter needs to be `Object` instead of `NodeUpdate` is because it's overriding the equals() method from the Object class, but you can still cast it to NodeUpdate within the method body to perform the equality comparison. 

Actaully, if u check other class equal method in `rest.li` code, u will find this is the common.
<img width="795" alt="image" src="https://github.com/linkedin/rest.li/assets/150380879/bf405b8d-68e5-466b-8f04-7eba4b8c4adf">

Reply:
makes sense. Sorry I missed it.